common/models/user.js

```diff
@@ -604,8 +604,8 @@ module.exports = function(User) {
     };
 
     UserModel.setter.password = function(plain) {
-      if (typeof plain !== 'string') {
-        return;
+      if (typeof plain === 'number') {
+        plain = plain.toString();
       }
```

Comment:
It's probably better to take advantage of the `validatePassword` function instead of just doing an early `return`... maybe by checking if `plain` is a string on line 610 below, which allows us to reach the `hashPassword` function where `validatePassword` is called:
```javascript
if (typeof plain === 'string' && plain.indexOf('$2a$') === 0 && plain.length === 60) {
  // ...
} else {
  this.$password = this.constructor.hashPassword(plain);
}
```
so if `plain` is of type `object` (e.g. , it will get passed along and validated:
```javascript
User.hashPassword = function(plain) {
  this.validatePassword(plain);
  // ...
};

User.validatePassword = function(plain) {
  if (typeof plain === 'string' && plain) {
    return true;
  }
  var err =  new Error('Invalid password: ' + plain);
  err.statusCode = 422;
  throw err;
};
```
at which point it will throw a better error message `Invalid password: { pw: 'abc123' }`, instead of what we currently get (`The User instance is not valid. Details: password can't be blank (value: undefined)`)

What do you think?